go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -114,26 +114,10 @@ func (c *addressManager) ListAddresses() []net.IP {
 type subscribeFn func() (bool, chan netlink.AddrUpdate, error)
 
 func (c *addressManager) Run(stopChan <-chan struct{}, doneWg *sync.WaitGroup) {
-	addrSubscribeOptions := netlink.AddrSubscribeOptions{
-		ErrorCallback: func(err error) {
-			klog.Errorf("Failed during AddrSubscribe callback: %v", err)
-			// Note: Not calling sync() from here: it is redudant and unsafe when stopChan is closed.
-		},
-	}
-
-	subscribe := func() (bool, chan netlink.AddrUpdate, error) {
-		addrChan := make(chan netlink.AddrUpdate)
-		if err := netlink.AddrSubscribeWithOptions(addrChan, stopChan, addrSubscribeOptions); err != nil {
-			return false, nil, err
-		}
-		// sync the manager with current addresses on the node
-		c.sync()
-		return true, addrChan, nil
-	}
 	c.addHandlerForPrimaryAddrChange()
 	doneWg.Add(1)
 	go func() {
-		c.runInternal(stopChan, subscribe)
+		c.runInternal(stopChan, c.getNetlinkAddrSubFunc(stopChan))
```

Comment:
self-note: the same channel is passed into runInternal and into addrSubscribeAt
the former simply returns if stop is called and latter has a goroutine which listens to that channel and then calls socket close..